No, don't use for calibration

Navid200 opened the pull request:
To reproduce the problem, go to Settings &#8722;> Less common settings &#8722;> Advanced Calibration &#8722;> Use Treatment BG values
Select "Ask me every time".

Now, go to the main screen and use the treatment menu to enter a blood glucose measurement.
When it asks you if you want to use the value for calibration, answer "No".  It will still use it for calibration.
You can clear the queue to undo calibration.

This PR fixes: https://github.com/NightscoutFoundation/xDrip/issues/1580

jamorham commented:
Thanks for this, appears at first glance as a well overdue todo. Can you confirm that after you say no you can then click on the blood test and calibrate later from it?

Navid200 commented:
@jamorham I just tested:
After answering no, it becomes a square marker on the main screen.  Tapping on it, shows time and value at the bottom as well as "BLOOD TEST" in blue at the bottom right of the screen.  Tapping on blood test, brings up a form that lets you do nothing, delete or calibrate.

I have to add one thing here:
I don't know why after tapping on no, nothing shows on screen.  One has to go to logs and come back to the main screen to see the square marker.  Or, one has to minimize xDrip and open it up again to see the square marker.
I don't remember this behavior.  

I just tested with never and the marker appears right away.  I need to figure out why.  I will find out and correct.

jamorham commented:
add a `staticRefreshBGChartsOnIdle()` call to your dialog lambda

Navid200 commented:
Thanks
But, why does the same problem not occur for never?!
Is this an obvious demonstration of me not being a programmer?  :o)

Navid200 commented:
@jamorham auto also has a dialog.  Why does it not happen with auto?

Navid200 commented:
The behavior of the marker is now consistent across all possible options.
I have tested them all.

jamorham commented:
Can you describe how the logic flow is changed by this and why this fix works? thanks

Navid200 commented:
@jamorham 
Regardless of which setting the user chooses on the form shown next, we need a marker to be placed.
![7](https://user-images.githubusercontent.com/51497406/166679041-f94bb894-823c-4acd-a90f-13629815d676.png)

But, we have repeat code for that and I am actually responsible for that:
https://github.com/NightscoutFoundation/xDrip/pull/1814/commits/e25d6e62ae47643426a70b28f67f80125631c66c
In fact @tolot27 caught that then and pointed it out.  But, I didn't get it.

When I first opened this PR yesterday, I had added a third instance of this code, which happened to be inside a dialog.  The consequence of that was that it caused an inconsistent behavior for the marker placement.

So, I have taken the part of the code that creates the marker and have moved it ahead of all the conditionals:
![Screenshot 2022-05-04 081856](https://user-images.githubusercontent.com/51497406/166679748-a0e1fabc-8eef-4937-9fba-6ed4c0b50dd6.png)
Now, all marker placements inside the conditionals are redundant.  So, I have removed those.

Now, the dialog has no impact on the marker.
I hope this makes sense.

Navid200 commented:
If you want to test this, please note:

1- After you enter a blood glucose value using the treatment menu, if you delete it, you cannot apply another one right away.  You need to wait for the next read cycle (5 minutes).  This limitation is not caused by this PR.  If you don't like that behavior, we will need another PR to change it.

2- If you are testing the "Auto" option, please pay attention to this memory:
![Screenshot 2022-05-04 100523](https://user-images.githubusercontent.com/51497406/166698618-78b76b71-009d-45ca-a869-eea6dab49d9f.png)
That is one month!  

That means after you test it and get a dialog asking you if you want to enable automatic calibration or not, and you respond, it's going to remember your response for one month and will never ask you again.
So, if you try again and don't see the question, again, please don't think this is caused by this PR.
The only way to test it again is to uninstall xDrip, which I don't recommend.  Do it on a test phone, not your mission mode phone.
